### ISSUE TITLE
iceprod2 - better handling of db function timeouts (Trac #871)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#865]
Add code to make sure the db methods return a value (or an error) within a timeout period.  Make sure we always send a reply if one is asked for.

Consider adding test cases for every db method to verify that it completes or returns an error instead of silently failing.

Migrated from https://code.icecube.wisc.edu/ticket/871
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:28", 
    "description": "Add code to make sure the db methods return a value (or an error) within a timeout period.  Make sure we always send a reply if one is asked for.\n\nConsider adding test cases for every db method to verify that it completes or returns an error instead of silently failing.", 
    "reporter": "david.schultz", 
    "cc": "ddelventhal", 
    "resolution": "fixed", 
    "_ts": "1550067088921308", 
    "component": "iceprod", 
    "summary": "iceprod2 - better handling of db function timeouts", 
    "priority": "critical", 
    "keywords": "", 
    "time": "2015-02-13T19:28:26", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150504T15:56:55: david.schultz changed milestone from "" to "IceProd 2.0"

**Comment by icecube-trac** [comment]
Trac update at 20150828T20:16:21: 
* david.schultz commented:

> In changeset 136984:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="136984"
> make a dbmethods decorator so every dbmethod can timeout and call the callback. fixes #865
> ```


* david.schultz changed status from "new" to "closed"
* david.schultz changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:28: 
* olivas commented:

> Milestone IceProd 2.0 deleted


* olivas changed milestone from "IceProd 2.0" to ""
